### PR TITLE
VNF deploy

### PR DESCRIPTION
**Pull request by dmitryintel**

### CODE REVIEW

File: osc-server/src/main/java/org/osc/core/broker/service/appliance/AddApplianceService.java [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+import org.osc.core.broker.service.dto.ApplianceDto;
+import org.osc.core.broker.service.persistence.OSCEntityManager;
+import org.osc.core.broker.service.request.BaseRequest;
+import org.osc.core.broker.service.response.BaseResponse;
+import org.osc.core.broker.service.validator.ApplianceDtoValidator;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+@Component(service = {AddApplianceServiceApi.class, AddApplianceService.class})
```
Comment (arvindn05) [inline osc-server/src/main/java/org/osc/core/broker/service/appliance/AddApplianceService.java]: we just need to expose this as AddApplianceServiceApi

File: osc-server/src/main/java/org/osc/core/broker/service/appliance/AddApplianceService.java [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+    private static final Logger log = Logger.getLogger(AddApplianceService.class);
+
+    @Reference
+    private ApiFactoryService apiFactoryService;
+
+    @Reference
+    private ApplianceDtoValidator applianceDtoValidator;
+
+    private String uploadPath;
```
Comment (arvindn05) [inline osc-server/src/main/java/org/osc/core/broker/service/appliance/AddApplianceService.java]: upload path is not used

File: osc-server/src/main/java/org/osc/core/broker/service/appliance/ImportApplianceSoftwareVersionService.java [hunk 3]
```diff
@@ -79,21 +83,21 @@ public BaseResponse exec(ImportFileRequest request, EntityManager em) throws Exc
...
 
             if (appliance == null) {
-                appliance = new Appliance();
-                appliance.setManagerType(imageMetadata.getManagerType());
-                appliance.setModel(imageMetadata.getModel());
-                appliance.setManagerSoftwareVersion(imageMetadata.getManagerVersion());
+                ApplianceDto applianceDto = new ApplianceDto(imageMetadata.getModel(),
+                                                             imageMetadata.getManagerType(),
+                                                             imageMetadata.getManagerVersion());
+                Long id = this.addApplianceService.addAppliance(applianceDto);
```
Comment (arvindn05) [inline osc-server/src/main/java/org/osc/core/broker/service/appliance/ImportApplianceSoftwareVersionService.java]: appliance id

File: osc-server/src/main/java/org/osc/core/broker/service/validator/ApplianceDtoValidator.java [hunk 4]
```diff
@@ -19,10 +19,41 @@
...
+
+@Component(service = ApplianceDtoValidator.class)
+public class ApplianceDtoValidator implements DtoValidator<ApplianceDto, Appliance> {
+
+    @Reference
+    private EntityManager em;
+
+    @Override
+    public void validateForCreate(ApplianceDto dto) throws Exception {
+        if (dto == null) {
```
Comment (arvindn05) [inline osc-server/src/main/java/org/osc/core/broker/service/validator/ApplianceDtoValidator.java]: dont need this check. we are going to fail with npe anyway if its null

File: osc-server/src/main/java/org/osc/core/broker/service/validator/ApplianceDtoValidator.java [hunk 5]
```diff
@@ -19,10 +19,41 @@
...
+        if (existingAppliance != null) {
+            throw new IllegalArgumentException("Appliance already exists for model: " + dto.getModel());
+        }
+
+        checkForNullFields(dto);
+    }
+
+    @Override
+    public Appliance validateForUpdate(ApplianceDto dto) throws Exception {
+        throw new NotImplementedException("Method validate for update not implemented yet");
```
Comment (arvindn05) [inline osc-server/src/main/java/org/osc/core/broker/service/validator/ApplianceDtoValidator.java]: throw UnsupportedOperationException...avoids dependency on commons lang

File: osc-server/src/test/java/org/osc/core/broker/service/appliance/ImportApplianceSoftwareVersionServiceTest.java [hunk 6]
```diff
@@ -45,12 +48,15 @@
 import org.mockito.Mock;
 import org.mockito.Mockito;
 import org.mockito.MockitoAnnotations;
+import org.mockito.invocation.InvocationOnMock;
+import org.mockito.stubbing.Answer;
 import org.osc.core.broker.model.entities.appliance.Appliance;
 import org.osc.core.broker.model.entities.appliance.ApplianceSoftwareVersion;
 import org.osc.core.broker.model.image.ImageMetadata;
 import org.osc.core.broker.model.plugin.ApiFactoryService;
 //import org.osc.core.broker.model.virtualization.VmwareSoftwareVersion;
```
Comment (arvindn05) [inline osc-server/src/test/java/org/osc/core/broker/service/appliance/ImportApplianceSoftwareVersionServiceTest.java]: remove commented code.

File: osc-service-api/src/main/java/org/osc/core/broker/service/dto/ApplianceDto.java [hunk 7]
```diff
@@ -29,6 +29,15 @@
     private String managerType;
     private String managerVersion;
 
+    public ApplianceDto() {
```
Comment (arvindn05) [inline osc-service-api/src/main/java/org/osc/core/broker/service/dto/ApplianceDto.java]: package private constructor.

**Comment by emanoelxavier** [comment]
I am closing this PR as i will be sending a new one from another branch with unit tests and addressing these comments.

**Pull request closed without merging**